lighthouse-core/gather/gatherers/dobetterweb/tags-blocking-first-paint.js

```diff
@@ -21,6 +21,21 @@ const Gatherer = require('../gatherer');
 /* global document,window */
 
 /* istanbul ignore next */
+function saveAsyncLinks() {
+  function checkForLinks() {
+    document.querySelectorAll('link').forEach(link => {
+      if (link.rel === 'preload' || link.disabled || link.async) {
+        window.__asyncLinks[link.href] = true;
+      }
+    });
+  }
+
+  window.__asyncLinks = window.__asyncLinks || {};
+  setInterval(checkForLinks, 100);
```

Comment:
also has the unfortunate wrinkle that if a blocking script occurs in the head before the stylesheet and the script takes longer than the stylesheet to load and execute the querySelector will return nothing and we will have missed the window to identify it as a preloaded stylesheet